src/Cronofy/UpsertEventRequestBuilder.cs

```diff
@@ -414,13 +439,25 @@ public UpsertEventRequest Build()
         {
             var request = new UpsertEventRequest
             {
-                EventId = this.eventId,
                 Summary = this.summary,
                 Description = this.description,
                 Start = GetEventTime("Start", this.startTime, this.startDate, this.startTimeZoneId),
                 End = GetEventTime("End", this.endTime, this.endDate, this.endTimeZoneId),
             };
 
+            if (string.IsNullOrEmpty(this.eventId) == false)
+            {
+                request.EventId = this.eventId;
+            }
+            else if (string.IsNullOrEmpty(this.eventUid) == false)
+            {
+                request.EventUid = this.eventUid;
+            }
+            else
+            {
+                throw new ArgumentException("Must supply either eventId or eventUid");
```

Comment:
This is encoding current server logic into client code. I've avoiding doing this, instead producing what the client asks for and allowing the server to validate it.

For example, there could be a future where providing both `event_id` and `event_uid` is meaningful, if the client allows that then people can easily update their code to accommodate it.